### DISCUSSION TITLE
Compatibility with Kotlin 2.0.0

### DISCUSSION THREAD

**Discussion by xmmendezy** [varabyte/kobweb#540] [Q&A] [answered]
Is there any plan to start updating to Kotlin 2.0.0? I am creating a new test project and having trouble using the latest Kotlin features.

**Comment by DennisTsar** [comment, marked as answer]
Support for Kotlin 2.0.0 is planned (see https://github.com/varabyte/kobweb/tree/feature/k2), but we unfortunately discovered an issue with K2 + Compose that is holding up the migration: https://youtrack.jetbrains.com/issue/KT-68463.

**Reply by xmmendezy** [reply]
Oh, I see, thank you very much for the information. I hadn't seen that branch. I also want to take this opportunity to thank you for the great work you are doing with Kobweb. I am eager to use it in production.

**Reply by bitspittle** [reply]
@xmmendezy 

This is *not* a recommendation, but if you wanted to set things up early, you can set your Kobweb version to `0.19.0-SNAPSHOT` (which is what the `feature/k2` branch generates).

At that point:
* be sure your kotlin and compose versions match [COMPATIBILITY.md](https://github.com/varabyte/kobweb/blob/feature/k2/COMPATIBILITY.md)
* review [the migration docs](https://github.com/varabyte/kobweb/blob/feature/k2/docs/k2-migration.md)

If you do this, please treat your project as if it is in a quarantine zone. Expect to run into the incremental compilation issue bug sometimes (at which point I think you'd need to quit and restart the running kobweb server?) and also understand that we are much more fast and loose with our SNAPSHOT versions than our stable releases.

That all said, if this is just a test project for now, maybe it's OK; and if you play around with it, it could help us if you let us know how it goes / find other issues we hadn't yet.

**Reply by xmmendezy** [reply]
Thank you, I will put it to the test and keep you updated with any news I have about it.
